docs/api-reference/types/message-builder.md

```diff
@@ -0,0 +1,121 @@
+# MessageBuilder
+
+[← Back to API Reference](../README.md) | [← Back to Table of Contents](../../README.md)
+
+`MessageBuilder` is a factory class in Reactive Domain that facilitates the creation of correlated messages, ensuring proper tracking of correlation and causation IDs across message flows.
+
+## Overview
+
+In event-sourced systems, tracking the flow of messages is crucial for debugging, auditing, and understanding causal relationships. The `MessageBuilder` factory provides a consistent way to create messages with properly set correlation and causation IDs.
+
+## Class Definition
+
+```csharp
+public static class MessageBuilder
+{
+    public static TMessage New<TMessage>(Func<TMessage> messageFactory) 
+        where TMessage : ICorrelatedMessage;
+        
+    public static TMessage From<TMessage>(ICorrelatedMessage source, Func<TMessage> messageFactory) 
+        where TMessage : ICorrelatedMessage;
+}
+```
+
+## Key Features
+
+- **Message Creation**: Simplifies the creation of new messages with unique IDs
+- **Correlation Tracking**: Automatically sets correlation IDs for tracking related messages
+- **Causation Tracking**: Establishes causation links between messages
+- **Type Safety**: Provides type-safe message creation through generic methods
+
+## Usage
+
+### Creating a New Message
+
+To create a new message that starts a new correlation chain:
+
+```csharp
+// Create a new command with a new correlation ID
+ICorrelatedMessage command = MessageBuilder.New(() => new CreateAccount(Guid.NewGuid()));
+```
+
+### Creating a Message from an Existing Message
+
+To create a message that continues an existing correlation chain:
+
+```csharp
+// Create a command with correlation information from an existing message
+ICorrelatedMessage existingCommand = // ... existing command
+ICorrelatedMessage newCommand = MessageBuilder.From(existingCommand, () => new DepositFunds(accountId, amount));
```

Comment:
This should be 
```
ICorrelatedMessage newCommand = MessageBuilder.From(existingCommand).Build(() => new DepositFunds(accountId, amount));
```
It seems the `Build` method is missing from the class definition above, so not surprising that it's missing here.